shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java

```diff
@@ -284,17 +283,14 @@ public boolean updateSubjectName(SubjectDTO subject) throws MicroServiceCommunic
 		}
 	}
 
-	@Override
-	public List<SimpleSubjectDTO> findAllSubjectsOfStudy(final Long studyId) {
-		return this.findAllSubjectsOfStudyId(studyId);
-	}
-
 	@Override
 	public List<SimpleSubjectDTO> findAllSubjectsOfStudyId(final Long studyId) {
 		List<SimpleSubjectDTO> simpleSubjectDTOList = new ArrayList<>();
-		List<SubjectStudy> opt = subjectStudyRepository.findByStudy(studyRepository.findById(studyId).orElse(null));
-		if (opt != null) {
-			for (SubjectStudy rel : opt) {
+		List<SubjectStudy> studySubjects = subjectStudyRepository.findByStudyId(studyId);
+		Study studyWithTags = studyRepository.findStudyWithTagsById(studyId);
+		studySubjects.stream().forEach(ss -> ss.setStudy(studyWithTags));
+		if (studySubjects != null) {
```

Comment:
Maybe return empty List after line 289 if studySubjects is null, to limit cyclomatic complexity ?

Reply:
well seen, done